packages/lib/src/application/App.ts

```diff
@@ -151,33 +134,45 @@ export default class App {
     initArgs?: string[],
   ): Promise<Contract> {
     if (!isEmpty(initArgs) && !initMethodName) initMethodName = 'initialize';
+    const implementation = await this.getImplementation(
+      packageName,
+      contractName,
+    );
     const proxy =
       initMethodName === undefined
-        ? await this._createProxy(packageName, contractName, proxyAdmin)
+        ? await this._createProxy(
+            packageName,
+            contractName,
+            implementation,
+            proxyAdmin,
+          )
         : await this._createProxyAndCall(
             contract,
             packageName,
             contractName,
+            implementation,
             proxyAdmin,
             initMethodName,
             initArgs,
           );
-    log.info(`${packageName} ${contractName} proxy: ${proxy.address}`);
+    Loggy.succeed(
+      `create-proxy-${implementation}`,
```

Comment:
Are we creating more than a single proxy at the same time? If so, why do we need to use the implementation as id and retrieve it earlier?